Fix critical warnings

MJoergen opened the pull request:
Some of the constraints were wrong. This PR fixes the incorrect constraints.

MJoergen committed:
Fix some critical warnings during build

MJoergen committed:
Add a audio clock constraint

Everything running on the audio clock was previously unconstrained.
Since the audio clock is running at less than 15 MHz (i.e. larger than
60 ns clock period), it's probably not a big deal.

I also replaced a "set_false_path" with "set_max_delay". The former is a
complete "ignore everything", whereas the latter still constrains the
routing delay.

MJoergen committed:
Fix incorrect clock constraints

lydon42 reviewed (commented):
Applied this over #812 and the build failed.

MJoergen committed:
Improve SDRAM constraints

This fix forces the I/O registers of the SDRAM controller to be placed
inside the I/O block (rather than in the main fabric of the FPGA).

The key is the "attribute iob".

However, in order to get full benefit, it's necessary to have explicit
registers of the "output enable", or rather "tristate enable".

Furthermore, the constraints are no longer necessary, because there is
nothing more to constrain.

lydon42 reviewed (approved):
My testbuild now looks good.

MJoergen committed:
SDRAM fix: Don't drive DQ during read
